### DISCUSSION TITLE
Problems with httpzip.

### DISCUSSION THREAD

**Discussion by Frida854** [Moddable-OpenSource/moddable#1099] [Q&A] [answered]
I have taken the project from 'moddable/examples/network/http/httpzip/'.
When I compile it, it runs as it should.
But when I take the original 'site.zip' and unzip and zip it again, and after I delete the project folders in respectively
'/Users/poulborg/Projects/moddable/build/bin/esp32/debug' and
'/Users/poulborg/Projects/moddable/build/tmp/esp32/debug',
when I compile again I get this error '/Users/poulborg/Projects/moddable/modules/files/zip/modZip.c (270) # Break: File: can't find path!'

It happens on Mac and linux, with latest updates.

If I move the original 'site.zip' back, and delete in affected folders,
it works again. So I can't make changes.

So how do I make changes to 'index.html' and make it work.
<img width="1098" alt="pic01" src="https://user-images.githubusercontent.com/18483582/235367953-54f8c277-e956-4a08-88b3-b576266f7945.png">

**Comment by phoddie** [comment, marked as answer]
There's [a note](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/files/files.md#class-zip) at the end of the ZIP file system module that explains what you are seeing:

> **Note**: The zip command line tool creates a directory named "test" at the root of the ZIP file. For example, the file at "test/example.txt" is accessed in the ZIP file as "test/example.txt" not "example.txt".

You should be able to access the site at `http://localhost/site/index.html` using the ZIP file you built.

On macOS, you can recreate the original site.zip file this way:

- In the `site` directory, Select All
- Select File > Compress
- Copy the `Archive.zip` that is created up one level to `site.zip`

That eliminates the "site" directory at the root of the ZIP. This process can be easily automated using a batch file with the `zip` command line tool.

Alternatively (and perhaps simplest!), keep the "site" directory prefix in the ZIP file and instead remap the file path using the `onRoute` callback in examples/http/httpzip/main.js:

```js
	onRoute(path) {
		if ("/" === path)
			path = '/index.html';
		return "/site" + path;
	}
```

**Reply by Frida854** [reply]
It seems like a dream once you know how it works.
Thank you for the good support, even on a Sunday.

**Reply by phoddie** [reply]
Glad it is working for you now! It is a very powerful feature for easily embedding a web site into a project.
